src/main/java/tr/com/srdc/cda2fhir/transform/CCDTransformerImpl.java

```diff
@@ -137,24 +137,20 @@ public void setIdGenerator(IdGeneratorEnum idGen) {
      * @param resourceProfileMap The mappings of default resource profiles to desired resource profiles. Used to set profile URI's of bundle entries or omit unwanted entries.
      * @return A FHIR Bundle that contains a Composition corresponding to the CCD document and all other resources but Patient that are referenced within the Composition.
      */
-    public Bundle transformDocument(ClinicalDocument cda, BundleType bundleType, String patientRef, Map<String, String> resourceProfileMap) {
-        // The default transformer will use this patient reference if it is set.
-        this.patientRef = new Reference("Patient/" + patientRef);
-
-        Bundle documentBundle =  transformDocument(cda);
-        if (documentBundle == null) return null;
-
+    public Bundle createTransactionBundle(Bundle bundle, BundleType bundleType, Map<String, String> resourceProfileMap, boolean initialBundle) {
         Bundle resultBundle = new Bundle();
```

Comment:
Changing the name to createTransactionBundle and then accepting a BundleType does not make sense to me.

Reply:
Well, a bundle doesn't have to be a transaction bundle. So this would be to create a transaction bundle from a non-transaction bundle.